ni/niveristand-communications-bus-template
# Unique custom device build for cloning.

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-communications-bus-template/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

The renamed source copy is now capable of being built into a unique custom device that runs in System Explorer.

- `Clone Template Main.vi` calls all the subVIs needed to create the cloned source.
- `Copy Template Source to New Location.vi` has been modified to save the `CD name` to a global.
- `Edit Paths in XML File.vi` has been modified to use the `CD name` global.
- `Edit Project File Build Spec Destinations.vi` has been added to edit the project to use the new CD name.
- `Edit VI Constants.vi` has been modified to use the `CD name` global.
- `Remove Test Folder From lvproj Temporarily.vi` has been added until we decide how to fix missing links to `niveristand-custom-device-testing-tools`.
- `Template Tool Global.vi` has the addition of `CD name`.
- `Update CD Name in XML File.vi` has been modified to use `CD name`. Please advise if we want to change **National Instruments::** to **NI::** at this point, or if we need to think about this more.

### Why should this Pull Request be merged?

This completes the second user story for cloning the Comms Bus Template.

### What testing has been done?

Testing on my dev VM includes:

- Run `Clone Template Main.vi`  to clone the source.
- Build the two support build specifications by hand: `Communication Bus Engine` and `Communication Bus System Explorer`.
- Build the two CD build specifications by hand: `Configuration Release` and `Engine Release`.
- Confirm files get copied to _C:\Users\Public\Documents\National Instruments\NI VeriStand 2020\Custom Devices_.
- Confirm the CD shows in System Explorer under  **Custom Devices>National Instruments** and appears to function.
